src/components/Widget/components/Conversation/components/Messages/components/Carousel/index.js

```diff
@@ -0,0 +1,171 @@
+import React, { useRef, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { addUserMessage, emitUserMessage } from 'actions';
+import { PROP_TYPES } from 'constants';
+import Arrow from 'assets/arrow.svg';
+
+import './styles.scss';
+
+const Carousel = (props) => {
+  const carousel = props.message.toJS();
+
+  const handleClick = (action) => {
+    if (!action || action.type !== 'postback') return;
+    const { chooseReply } = props;
+    chooseReply(action.payload, action.title);
+  };
+
+  const scrollContainer = useRef();
+  const [leftButton, setLeftButton] = useState(false);
+  const [rightButton, setRightButton] = useState(true);
+
+  const handleScroll = () => {
+    const current = scrollContainer.current;
+    if (scrollContainer.current.scrollLeft > 0) {
+      setLeftButton(true);
+    } else {
+      setLeftButton(false);
+    }
+    if (current.clientWidth === current.scrollWidth - current.scrollLeft) {
+      setRightButton(false);
+    } else {
+      setRightButton(true);
+    }
+  };
+
+  const handleLeftArrow = () => {
+    scrollContainer.current.scrollTo({
+      left: scrollContainer.current.scrollLeft - 230,
+      behavior: 'smooth'
+    });
+  };
+
+  const handleRightArrow = () => {
+    scrollContainer.current.scrollTo({
+      left: scrollContainer.current.scrollLeft + 230,
+      behavior: 'smooth'
+    });
+  };
+
+  return (
+    <React.Fragment>
+      <div className="rw-carousel-container" ref={scrollContainer} onScroll={() => handleScroll()}>
+        {carousel.elements.map((carouselCard, index) => {
+          const defaultActionUrl =
+            carouselCard.default_action && carouselCard.default_action.type === 'web_url'
+              ? carouselCard.default_action.url
+              : null;
+          return (
+            <div className="rw-carousel-card" key={index}>
+              <a
+                href={defaultActionUrl}
+                target="_blank"
```

Comment:
minor/improvement: there is a metadata "linkTarget" that define where links should open (either the current windows or a new tabs) maybe use it here ? so all links of the webchat are affected by the metadata